DashAISoftware/DashAI
# Runner Endpoint

# Summary

- Runner endpoint (path _/runner/?run_id=_).
- Test test_runner.
- Run endpoint: Delete trained run file from file system.
- Run model: Change run_name and run_description to name and description.
- Test Dataset: Delete all the created dataset.

This PR depends on https://github.com/DashAISoftware/DashAI/pull/78


## Type of change



- Back end new feature.

## Changes

Add runner endpoint (_/runner/_) to the app. It's only available the post method which train and test the specified run.

The runner endpoint follow this pipeline:
- Get the Run, Experiment and Dataset from the DB.
- Instantiate the run with it's parameters.
- Instantiate the Dataset. 
- Train the run.
- Get the train and test results using the metrics associated with the Task associated to the Experiment. (TBD)
- Save the trained run.

Add test _test_runner_ to check the behavior of the endpoint.
- The test depends on dataset endpoint.
- The test mocks the registry and use own Dummy components.

Run endpoint:
- If the run was trained (i.e. RunStatus = FINISHED) tries to delete the saved run file.
- If the run was trained and the file does not exists it throws a HTTP Exception.

Run database model: 
- Change run_name and run_description to name and description.
- Refactor the change in the Run endpoint and in the test_runs files.

Test Dataset: 
- Delete all the created dataset. The remaining datasets could cause errors in other tests that uses datasets (like test_runner)



## How to Test



In the root of the proyect you can run `pytest .\tests\back\api\test_runner.py ` to execute the test.
It's also good idea to run `pytest .` to check test_runs and test_datasets are still working.
